pkgs/development/python-modules/plyer/default.nix

```diff
@@ -0,0 +1,77 @@
+{ lib, buildPythonPackage, fetchFromGitHub, mock, python3Packages, python, stdenv }:
+
+buildPythonPackage rec {
+
+  pname = "plyer";
+  version = "2.0.0";
+
+  src = fetchFromGitHub {
+    owner = "kivy";
+    repo = "plyer";
+    rev = version;
+    sha256 = "15z1wpq6s69s76r6akzgg340bpc21l2r1j8270gp7i1rpnffcjwm";
+  };
+
+  propagatedBuildInputs = [ python3Packages.keyring ];
+
+  checkInputs = [ mock ];
+  preCheck = ''
+    HOME=$(mktemp -d)
+    mkdir -p $HOME/.config/
+    touch $HOME/.config/user-dirs.dirs
+    mkdir -p $HOME/Pictures/
+  '';
+  checkPhase = ''
+    runHook preCheck
+    ${python.interpreter} -m unittest discover \
+      --start-directory plyer/tests \
+      --top-level-directory .
+    runHook postCheck
+  '';
+
+  postPatch = ''
+    # Fixup this file to
+    # - not assume ~/.config/user-dirs.dirs exists
+    # - not assume it actually has stuff in it if it does exist
+    # - return the path under HOME if user-dirs has no useful info.
+    # - use the right name for DOCUMENTS according to the spec
+    substituteInPlace plyer/platforms/linux/storagepath.py \
+      --replace \
+        "except KeyError:" "except (KeyError, IndexError, FileNotFoundError):" \
+      --replace \
+        "return PATHS[name]" "return self._get_home_dir() + PATHS[name]" \
+      --replace \
+        $'directory = self._get_from_user_dirs("DOCUMENT")' \
+        $'directory = self._get_from_user_dirs("DOCUMENTS")'
+
+    rm -r examples
+    # Test assumes dbus is not installed, it fails and is not very robust.
+    substituteInPlace plyer/tests/test_notification.py --replace \
+      "    def test_notification_notifysend(self):" \
+      $'    @unittest.skip("")\n    def test_notification_notifysend(self):'
+    # this fails to produce a uniqueid.id, but I would not be able to explain why.
+    # It does work outside the nix build.
+    substituteInPlace plyer/tests/test_uniqueid.py --replace \
+      "    def test_uniqueid(self):" \
+      $'    @unittest.skip("")\n    def test_uniqueid(self):'
+    # remove all the wifi stuff. Depends on a python wifi module that has not been
+    # updated since 2016
+    find -iname "wifi*" -exec rm {} \;
+    substituteInPlace plyer/__init__.py \
+      --replace "wifi = Proxy('wifi', facades.Wifi)" "" \
+      --replace "'wifi'" ""
+    substituteInPlace plyer/facades/__init__.py \
+      --replace "from plyer.facades.wifi import Wifi" ""
+  '';
+
+  pythonImportsCheck = [ "plyer" ];
+
+  meta = with lib; {
+    description =
+      "Plyer is a platform-independent api to use features commonly found on various platforms";
+    homepage = "https://github.com/kivy/plyer";
+    license = licenses.mit;
+    maintainers = with maintainers; [ rski ];
+    broken = stdenv.isDarwin;
```

Comment:
This should probably be converted into a disabled if it never will work on darwin.

Reply:
the plyer package is supposed to provide cross platform APIs that include darwin. I don't have access to macos hardware to fix it though. If anyone wants to pick it up they can though.

Reply:
```suggestion
```
If there is no concrete build failure and you just did not test it on darwin we don't set this flag.

Reply:
it definitely is broken on Darwin, see https://github.com/NixOS/nixpkgs/pull/136902#issuecomment-913862312